reconcile/vpc_peerings_validator.py

```diff
@@ -14,10 +16,136 @@
 )
 from reconcile.status import ExitCodes
 from reconcile.utils import gql
+from reconcile.utils.aws_api import AWSApi
 
 QONTRACT_INTEGRATION = "vpc-peerings-validator"
 
 
+def validate_no_cidr_overlap(
+    query_data: VpcPeeringsValidatorQueryData,
+) -> bool:
+    clusters: list[ClusterV1] = query_data.clusters or []
+
+    for cluster in clusters:
+        peerings_enteries_dict = {}  # type: ignore[var-annotated]
+        if cluster.peering:
+            for peering in cluster.peering.connections:
+                if peering.provider == "account-vpc-mesh":
+                    aws_account_uid = peering.account.uid  # type: ignore[union-attr]
+                    settings = queries.get_secret_reader_settings()
+                    accounts = queries.get_aws_accounts(uid=aws_account_uid)
+                    awsapi = AWSApi(1, accounts, settings=settings, init_users=False)
+                    tags = peering.tags or "{}"  # type: ignore[union-attr]
+                    mesh_results = awsapi.get_vpcs_details(accounts[0], tags)
+                    for mesh_result in mesh_results:
+                        vpc_peering_info = {
+                            "provider": peering.provider,
+                            "vpc_peering": {
+                                "vpc_name": mesh_result["vpc_id"],
+                                "cidr_block": mesh_result["cidr_block"],
+                            },
+                        }
+                        create_vpc_peering_dict(
+                            vpc_peering_info,
+                            peerings_enteries_dict,
+                            cluster.network.vpc,  # type: ignore[union-attr]
+                        )
```

Comment:
We can directly append vpc_peering_info to the result.
```suggestion
                        vpc_peering_info = {
                            "provider": peering.provider,
                            "vpc_name": mesh_result["vpc_id"],
                            "cidr_block": mesh_result["cidr_block"],
                        }
                        peerings_enteries.append(vpc_peering_info)
```